### PR TITLE
247797 ICDS Tableau Reports Changes

### PR DESCRIPTION
**Pull request by kkrampa**

### CODE REVIEW

File: corehq/apps/locations/static/locations/js/location_drilldown.async.js [hunk 1]
```diff
@@ -79,7 +86,14 @@ function LocationSelectViewModel(hierarchy, default_caption, auto_drill, loc_fil
 
     // load location hierarchy and set initial path
     this.load = function(locs, selected) {
-        this.root(new model.func({name: '_root', children: locs}, this));
+        if (!initialData) {
+            initialData = {
+                locs: locs,
+                selected: selected
```
Comment (dimagimon) [inline corehq/apps/locations/static/locations/js/location_drilldown.async.js]: Missing trailing comma. (comma-dangle)

File: custom/icds_reports/static/tableau_app.js [hunk 2]
```diff
@@ -111,7 +202,7 @@ function updateViz(marks) {
         // default the sheet to navigate to to the current one
         currentSheet = history.state.sheetName,
         newSheet = currentSheet;
-        newParams = {};
+        var newParams = {};
```
Comment (dimagimon) [inline custom/icds_reports/static/tableau_app.js]: Expected indentation of 4 space characters but found 8. (indent)

File: custom/icds_reports/views.py [hunk 3]
```diff
@@ -1,48 +1,70 @@
...
+    @property
+    def domain(self):
+        return self.kwargs['domain']
+
+    @property
+    def couch_user(self):
+        return self.request.couch_user
+
+    def get_context_data(self, **kwargs):
+        location_type_code, user_location_id, state_id, district_id, block_id = _get_user_location(self.couch_user, self.domain)
```
Comment (dimagimon) [inline custom/icds_reports/views.py]: E501 line too long (128 > 115 characters)

**Comment by kkrampa** [comment]
@esoergel Could you look at 	f254d1c? @sheelio asked me about "Not auto-select child locations if there is only one available child".

**Comment by kkrampa** [comment]
@sheelio I added requested filters.

**Comment by sheelio** [comment]
@kkrampa 

Can you look at the setUpInitialTableauParams function in the tableau_app.js.  It might need to be replaced or modified to account for your changes.

File: corehq/apps/locations/static/locations/spec/location_drilldown_spec.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*eslint-env mocha */
+/*global _ LocationSelectViewModel */
+
+
+var LOAD_LOCS_URL = '/locations/api/';
```
Comment (dimagimon) [inline corehq/apps/locations/static/locations/spec/location_drilldown_spec.js]: 'LOAD_LOCS_URL' is defined but never used (no-unused-vars)

**Comment by kkrampa** [comment]
@sheelio I updated PR

@esoergel Could you look at f254d1c? (or point better person to do this)

**Comment by sheelio** [comment]
@esoergel 

Are you able to deploy this to staging?  I'm having a hard time getting it to work locally on my machine.

**Comment by esoergel** [comment]
@kkrampa This looks good.  I'm surprised that that model has the ability to disable drilldown, but no way to actually specify that parameter.  Your approach here looks appropriate.

@sheelio Do we actually _want_ this filter to ever drilldown automatically?  The behavior feels frustrating.  This filter is used exclusively in CommTrack reports and custom stuff.  I wonder if we can just change it to never drilldown automatically.

**Comment by sheelio** [comment]
I find the behaviour pretty frustrating, but I was nervous about changing functionality for existing consumers of this filter.

**Comment by sheelio** [comment]
@kkrampa 

Did a bit of testing and some feedback:
- It doesn't look like the back and forward buttons synchronize the changes back to the filter UI.  I.e. I can change the date to November 2015 hit Apply, then change it to October 2015 and hit apply again.  Then I can press the Back button - the report updates back to October 2015, but the filters still show November 2015.
- Is there code that sets the initial values of the location parameters to be the user's assigned location.  I don't see you've added or changed that?
- We we should remove lines 146-148 from the tableau_app.js (since your code already sets the default month)

**Comment by kkrampa** [comment]
@sheelio Updated. Location is set here https://github.com/dimagi/commcare-hq/pull/15072/files#diff-c3d321a7b730e72d222c9be8120c6c7fR175 and they are get from the server here: https://github.com/dimagi/commcare-hq/pull/15072/files#diff-4a97252aba628a8783dc1ded82fb3c06R79

**Comment by sheelio** [comment]
@kkrampa Thanks! 

I'm also noticing that the filters on the left side don't properly wrap their labels (i.e. the label is hidden by the dropdown.  Is there a way to better format that? (@esoergel)

**Comment by sheelio** [comment]
@kkrampa 

A few more changes:
- It looks like the filter only lets you choose up to a block.  We need the ability to choose down to an AWC
- Does the filter have the behaviour where choosing a given AWC will also change the displayed view (see the section on special AWC ID handling in this section - https://docs.google.com/document/d/198taeD0MP1UXxUAUArKb-lFg97yF1tXz0Y7VBYmAflQ/edit#heading=h.ka8xvlnxguxz)?

**Comment by kkrampa** [comment]
Updated

**Comment by sheelio** [comment]
@esoergel 

I think this is good to go from my end, but can you hold off on merging this.  I have to stage this with a few other change.es

**Comment by esoergel** [comment]
@sheelio Sure thing.  I reviewed the changes outside of `custom/` again yesterday, and they're good to go on my end.

**Comment by sheelio** [comment]
@kkrampa 

I'm going to close this PR.  I've made some changes in another branch and I'd like to start making changes based on that.  
https://github.com/dimagi/commcare-hq/tree/ss/icds-tableau-filters

**Pull request closed without merging**